m3/src/main/java/com/uber/m3/tally/m3/M3Reporter.java

```diff
@@ -556,11 +556,12 @@ public void run() {
                 }
             }
 
-            state.set(ProcessorState.SHUTDOWN);
-
             LOG.warn("Processor shutting down");
 
-            shutdown();
+            if (isShutdown.get()) {
```

Comment:
This is incorrect -- processor had to do its own shutdown regardless whether the Reporter is shutdown or not -- it should skip drainage, unless reporter is also shutdown

Reply:
Good point, fixed.